### ISSUE TITLE
How to implement DTCWT to 1-D signals?

### ISSUE THREAD

**Issue by QuasiLegendre** [fbcotter/pytorch_wavelets#35]
How to implement DTCWT to 1-D signals? Also, does DTCWT has advantage over DWT in signal processing?

**Comment by fbcotter** [comment]
Hi @QuasiLegendre, it's not currently available in this repo but it wouldn't be too hard to add if it would be of use. As for the advantages of the DTCWT over the DWT - yes there are indeed! I'll refer you to the journal paper here: https://eeweb.engineering.nyu.edu/iselesni/pubs/CWT_Tutorial.pdf, in particular note the 4 problems mentioned in the introduction.

However, the DTCWT does add some complexity too, so it's hard to say if it's worth the extra effort. What's the use case you're thinking of?

**Comment by QuasiLegendre** [comment]
Thanks for your sincere reply! My use case is to find a optimized wavelet filtering/smoothing method for my particular 1-D real signal for signal anomaly detection in the next step(LSTM/GRU or 1-D CNN).  
![image](https://user-images.githubusercontent.com/33156295/146337657-665ec357-cd57-49b0-883e-d949d46c1210.png)
The signal is shown above, the red-labeled part is abnormal signal, while the blue part is referred as normal signal. The x axis means time, and y axis is voltage.

**Comment by fbcotter** [comment]
If you want to get good localisation of the signal then yes perhaps the DTCWT will be a good one to use. I can make a 1D version of it for you but it may take a couple of days. In the meantime, I'd suggest first trying with the DWT to see how well it works though.

**Comment by QuasiLegendre** [comment]
I will take your advice and try DWT in several days. Thank you the recommendations!

**Comment by lostmsu** [comment]
@fbcotter any plans to release 1.3 with 1D DWT?

**Comment by fbcotter** [comment]
Do you mean a 1D DTCWT? There already should be a 1D DWT in there. I see I haven't made docs for it though, so maybe you just didn't see it.
```python
from pytorch_wavelets import DWT1DForward, DWT1DInverse
```

**Comment by lostmsu** [comment]
@lostmsu no, I mean PyPi only has 1.2, so to get 1.3 one currently needs to install from source.

**Comment by naba89** [comment]
Hey, did you ever get around to implementing the 1D DTCWT?

**Comment by Tokenmw** [comment]
Hello, I would like to ask if you have implemented DTCWT to 1-D signals? Thank you so much.
